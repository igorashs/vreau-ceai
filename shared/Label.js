import styled from 'styled-components';

export const Label = styled.label`
  font-weight: 500;

  ${({ error }) => error && 'color: var(--text-danger);'}

  ${({ success, error }) =>
    success && !error && 'color: var(--accent-text-dark); font-weight: 400'}
`;
