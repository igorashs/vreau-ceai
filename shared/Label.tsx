import styled from 'styled-components';

interface LabelProps {
  error?: boolean;
  success?: boolean;
}

export const Label = styled.label<LabelProps>`
  font-weight: 500;

  ${({ error }) => error && 'color: var(--text-danger);'}

  ${({ success, error }) =>
    success && !error && 'color: var(--accent-text-dark); font-weight: 400'}
`;
