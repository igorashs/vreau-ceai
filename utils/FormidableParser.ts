import * as validator from '@/utils/validator';
import FormidableNameSpace, { Files, File, IncomingForm } from 'formidable';
import { NextApiRequest } from 'next';
import { Formidable, ProductFields } from 'types';
import { productMessages } from './validator/schemas/product';
import { NextApiSessionRequest } from './withSession';

type ParsedData = {
  fields: Partial<ProductFields>;
  files: Files & { src?: File };
};

type HandleProductForm = (
  req: NextApiRequest | NextApiSessionRequest,
  options: Partial<FormidableNameSpace.Options>,
  allowedFileTypes: string[],
) => Promise<ParsedData>;

const handleProductForm: HandleProductForm = async (
  req,
  options,
  allowedFileTypes,
) => {
  const data: ParsedData = await new Promise((resolve, reject) => {
    const form: Formidable = new IncomingForm(options);

    form.onPart = (part) => {
      if (part.filename) {
        // get file extension
        const fileType = part.mime?.split('/').pop() ?? '';

        // validate allowed file types
        if (!allowedFileTypes.includes(fileType)) {
          form.emit(
            'error',
            validator.createValidationError({
              message: productMessages.src.invalid,
              key: 'src',
            }),
          );
        }
      }

      form.handlePart(part);
    };

    form.parse(req, (err, fields, files) => {
      if (err) {
        // custom error on maxFileSize
        if (err.message.includes('maxFileSize')) {
          reject(
            validator.createValidationError({
              message: productMessages.src.max,
              key: 'src',
            }),
          );
        } else {
          reject(err);
        }
      }

      resolve({ fields, files });
    });
  });

  return data;
};

export default {
  handleProductForm,
};
