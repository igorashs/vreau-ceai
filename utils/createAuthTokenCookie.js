import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createAuthTokenCookie = (payload) => {
  const authToken = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  return cookie.serialize('auth-token', authToken, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });
};
