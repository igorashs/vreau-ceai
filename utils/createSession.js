import jwt from 'jsonwebtoken';
import cookie from 'cookie';

const EXPIRE_TIME = 24 * 60 * 60;

export const createSession = (payload) => {
  const token = jwt.sign(payload, process.env.AUTH_PRIVATE_KEY, {
    expiresIn: EXPIRE_TIME
  });

  const [header, jwtPayload, signature] = token.split('.');
  const session = header + '.' + jwtPayload;

  const sessionCookie = cookie.serialize('session', session, {
    secure: process.env.NODE_ENV !== 'development',
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  const authCookie = cookie.serialize('auth', signature, {
    secure: process.env.NODE_ENV !== 'development',
    httpOnly: true,
    sameSite: 'strict',
    path: '/',
    maxAge: EXPIRE_TIME
  });

  return [sessionCookie, authCookie];
};
