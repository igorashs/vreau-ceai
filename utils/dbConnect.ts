import mongoose from 'mongoose';

const dbConnect = async () => {
  if (mongoose.connection.readyState >= 1) {
    return;
  }

  await mongoose.connect(process.env.MONGODB_URI as string, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useCreateIndex: true,
    useFindAndModify: false,
  });
};

export default dbConnect;
