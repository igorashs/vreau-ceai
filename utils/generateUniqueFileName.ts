import { nanoid } from 'nanoid';

const generateUniqueFileName = (fileName: string) => {
  // grab file extension
  const matches = fileName?.match(/\..+$/);

  // create new file name
  const uniqueFileName = nanoid() + (matches && matches[0]);

  return uniqueFileName;
};

export default generateUniqueFileName;
