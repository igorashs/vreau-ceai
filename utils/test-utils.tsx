import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
