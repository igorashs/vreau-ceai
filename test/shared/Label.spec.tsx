import { render } from '@/utils/test-utils';
import Label from '@/shared/Label';

describe('Label', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(<Label>heyo</Label>);

    expect(getByText('heyo')).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('error prop', () => {
    it('renders with error style', () => {
      const { container } = render(<Label error>heyo</Label>);

      expect(container.firstChild).toHaveStyleRule(
        'color',
        'var(--text-danger)',
      );
    });
  });

  describe('success prop', () => {
    it('renders with success style', () => {
      const { container } = render(<Label success>heyo</Label>);

      expect(container.firstChild).toHaveStyleRule(
        'color',
        'var(--accent-text-dark)',
      );
      expect(container.firstChild).toHaveStyleRule('font-weight', '400');
    });

    it('renders with error style', () => {
      const { container } = render(
        <Label success error>
          heyo
        </Label>,
      );

      expect(container.firstChild).toHaveStyleRule(
        'color',
        'var(--text-danger)',
      );
    });
  });
});
