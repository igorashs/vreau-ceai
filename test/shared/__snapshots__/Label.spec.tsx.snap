// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
.c0 {
  font-weight: 500;
}

<label
  class="c0"
>
  heyo
</label>
`;
